cplusplus/gen-operators.py

```diff
@@ -128,6 +132,8 @@ def generate_operation(operation_name, declaration_only=False, indent=''):
             result += f'\n{indent} * @return {details["blurb"]}.'
 
         result += f'\n{indent} */\n'
+        if intro.flags & _OPERATION_DEPRECATED:
+            result += f'{indent}G_DEPRECATED\n'
```

Comment:
Note that this uses `G_DEPRECATED` instead of `VIPS_DEPRECATED` since the `extern` specifier cannot be applied to class members.